kevm_pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -107,56 +112,33 @@ def gen_spec_modules(kevm: KEVM, spec_module_name: str) -> str:
     return kevm.pretty_print(spec_defn)
 
 
-def solc_to_k(kevm: KEVM, contract_file: Path, contract_name: str, generate_storage: bool):
+def solc_to_k(kevm: KEVM, contract_json: Dict, contract_name: str, generate_storage: bool, foundry: bool = False) -> KFlatModule:
 
-    solc_json = solc_compile(contract_file)
-    contract_json = solc_json['contracts'][contract_file.name][contract_name]
-    storage_layout = contract_json['storageLayout']
     abi = contract_json['abi']
-    hashes = contract_json['evm']['methodIdentifiers']
-    bin_runtime = '0x' + contract_json['evm']['deployedBytecode']['object']
-
-    # TODO: add check to kevm:
-    # solc version should be >=0.8.0 due to:
-    # https://github.com/ethereum/solidity/issues/10276
+    hashes = contract_json['evm']['methodIdentifiers'] if not foundry else contract_json['methodIdentifiers']
+    bin_runtime = '0x' + (contract_json['evm']['deployedBytecode']['object'] if not foundry else contract_json['deployedBytecode']['object'])
 
     contract_sort = KSort(f'{contract_name}Contract')
 
-    storage_sentences = generate_storage_sentences(contract_name, contract_sort, storage_layout) if generate_storage else []
+    storage_sentences = []
+    if generate_storage:
+        storage_layout = contract_json['storageLayout']
+        storage_sentences = generate_storage_sentences(contract_name, contract_sort, storage_layout)
     function_sentences = generate_function_sentences(contract_name, contract_sort, abi)
```

Comment:
I am not sure about this. On my foundry example the `json` files don't have `storageLayout` entry, so I presume this check should only apply when `not foundry`.

Reply:
You have to call `foundry` with eth arguments in the `Makefile`. If you do that, do you get the correct entries?